src/Tribe/Repositories/Traits/Post_Tickets.php

```diff
@@ -0,0 +1,292 @@
+	 * ticket costs in place of their own cost meta.
+	 *
+	 * @since TBD
+	 *
+	 * @param float|array $value       The cost to use for the comparison; in the case of `BETWEEN`, `NOT BETWEEN`,
+	 *                                 `IN` and `NOT IN` operators this value should be an array.
+	 * @param string      $operator    Teh comparison operator to use for the comparison, one of `<`, `<=`, `>`, `>=`,
+	 *                                 `=`, `BETWEEN`, `NOT BETWEEN`, `IN`, `NOT IN`.
+	 * @param string      $symbol      The desired currency symbol or symbols; this symbol can be a currency ISO code,
+	 *                                 e.g. "USD" for U.S. dollars, or a currency symbol, e.g. "$".
+	 *                                 In the latter case results will include any event with the matching currency
+	 *                                 symbol, this might lead to ambiguous results.
+	 *
+	 * @throws Tribe__Repository__Usage_Error If the comparison operator is not supported of is using the `BETWEEN`,
+	 *                                        `NOT BETWEEN` operators without passing a two element array `$value`.
+	 */
+	public function filter_by_cost( $value, $operator = '=', $symbol = null ) {
+		$repo = $this;
+
+		// If the repo is decorated, use that.
+		if ( ! empty( $repo ) ) {
+			$repo = $this->decorated;
+		}
+
+		$operators = [
+			'<',
+			'<=',
+			'>',
+			'>=',
+			'=',
+			'!=',
+			'BETWEEN',
+			'NOT BETWEEN',
+			'IN',
+			'NOT IN',
+		];
+		if ( ! in_array( $operator, $operators, true ) ) {
+			throw Tribe__Repository__Usage_Error::because_this_comparison_operator_is_not_supported( $operator, 'filter_by_cost' );
+		}
+
+		if (
+			in_array( $operator, [ 'BETWEEN', 'NOT BETWEEN' ], true ) 
+			&& ! (
+				is_array( $value ) 
+				&& 2 === count( $value )
+			) 
+		) {
+			throw Tribe__Repository__Usage_Error::because_this_comparison_operator_requires_an_value_of_type( $operator, 'filter_by_cost', 'array' );
+		}
+
+		if ( in_array( $operator, [ 'IN', 'NOT IN' ], true ) ) {
+			$value = (array) $value;
+		}
+
+		$operator_name = Tribe__Utils__Array::get( Tribe__Repository::get_comparison_operators(), $operator, '' );
+		$prefix        = str_replace( '-', '_', 'by_cost_' . $operator_name );
+
+		global $wpdb;
+
+		// Join to the meta that relates tickets to events.
+		$repo->join_clause( "JOIN {$wpdb->postmeta} {$prefix}_ticket_event 
+			ON ( 
+				{$prefix}_ticket_event.meta_value = {$wpdb->posts}.ID
+				AND {$prefix}_ticket_event.meta_key REGEXP '^_tribe_.*_for_event$' 
+			)" );
+
+		$price_regexp_frags = [
+			// PayPal and WooCommerce tickets.
+			'_price',
+			// Easy Digital Downloads tickets.
+			'edd_price',
+		];
+		$price_regexp       = '^(' . implode( '|', $price_regexp_frags ) . ')$';
+
+		// Join to the ticket cost meta, allow for RSVP tickets too that have no price.
+		$repo->join_clause( $wpdb->prepare( "LEFT JOIN {$wpdb->postmeta} {$prefix}_ticket_cost 
+			ON (
```

Comment:
* :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: fba21d1a68c4a14db97b615545cf323e</sub>